arangod/Graph/GraphOperations.cpp

```diff
@@ -605,22 +589,122 @@ OperationResult GraphOperations::updateEdge(const std::string& definitionName,
                                             boost::optional<TRI_voc_rid_t> rev,
                                             bool waitForSync, bool returnOld,
                                             bool returnNew, bool keepNull) {
+  std::string fromCollectionName;
```

Comment:
Can we extract this validation code into a seperate helper function?
update and replace seem to use identical versions of it.
Maybe you can return a pair<result, trx> in this helper function and if result is good ust trx for the modify operation